File: plan-api/src/main/java/com/kpi/dyploma/planapi/model/plan/FinancialPlan.java
package com.kpi.dyploma.planapi.model.plan;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.kpi.dyploma.planapi.model.portfolio.Portfolio;
import jakarta.persistence.*;
import lombok.Data;
import lombok.ToString;

import java.util.List;

@Entity
@Data
@Table(name = "financial_plan")
public class FinancialPlan {
    private String name;
    @Id
    @GeneratedValue
    private Integer id;
    private Integer userId;
    private Integer startAge;
    private Integer pensionAge;
    private Integer governmentPension;
    private Integer endAge;
    private Integer riskProfile;
    private Float paceIncreaseOfExpense;
    private Float paceIncreaseOfIncome;
    private Integer mediumIncomeOfPortfolio;
    private Integer activeIncome;
    private Integer passiveIncome;
    private Integer activeOutcome;
    private Integer additionalOutcome;
    @OneToMany(mappedBy = "financialPlan", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<FinancialGoal> financialGoals;
    @OneToOne(mappedBy = "financialPlan", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private PlanAssets planAssets;
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "portfolio_id", referencedColumnName = "id")
    private Portfolio portfolio;
}

File: financial-assistant-ui/src/app/models/plan/financial-plan.ts
import {FinancialGoal} from "./financial-goal";
import {PlanAssets} from "./plan-assets";
import {Portfolio} from "./portfolio/portfolio";

export class FinancialPlan {
  public id: number;
  public userId: number;
  public startAge: number;
  public pensionAge: number;
  public governmentPension: number;
  public endAge: number;
  public riskProfile: number;
  public paceIncreaseOfExpense: number;
  public paceIncreaseOfIncome: number;
  public mediumIncomeOfPortfolio: number;
  public activeIncome: number;
  public passiveIncome: number;
  public activeOutcome: number;
  public additionalOutcome: number;
  public financialGoals: FinancialGoal[];
  public planAssets: PlanAssets;
  public portfolio: Portfolio;

}
